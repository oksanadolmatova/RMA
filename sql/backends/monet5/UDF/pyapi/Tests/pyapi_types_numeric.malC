

a:= bat.new(:int);
bat.append(a,1:int);
bat.append(a,2:int);
bat.append(a,3:int);
bat.append(a,4:int);
b:= bat.new(:int);
bat.append(b,4:int);
bat.append(b,3:int);
bat.append(b,2:int);
bat.append(b,1:int);
result:bat[:int] := pyapi.eval(nil:ptr, "return(arg1*sum(arg2) * arg3)", a, b, 2);
io.print(result);

# bits with null value
bbit:= bat.new(:bit);
bat.append(bbit,1:bit);
bat.append(bbit,0:bit);
bat.append(bbit,1:bit);
bat.append(bbit,0:bit);
bat.append(bbit,nil:bit);
rbit:bat[:int] := pyapi.eval(nil:ptr,"return(arg1)",bbit);
io.print(rbit);

# bytes with null value
bbte:= bat.new(:bte);
bat.append(bbte,42:bte);
bat.append(bbte,84:bte);
bat.append(bbte,111:bte);
bat.append(bbte,0:bte);
bat.append(bbte,nil:bte);
rbte:bat[:int] := pyapi.eval(nil:ptr,"return(arg1)",bbte);
io.print(rbte);

# short with null value
bsht:= bat.new(:sht);
bat.append(bsht,42:sht);
bat.append(bsht,82:sht);
bat.append(bsht,0:sht);
bat.append(bsht,240:sht);
bat.append(bsht,nil:sht);
rsht:bat[:int] := pyapi.eval(nil:ptr,"return(arg1)",bsht);
io.print(rsht);

# int with null value
bint:= bat.new(:int);
bat.append(bint,1804289383:int);
bat.append(bint,846930886:int);
bat.append(bint,1681692777:int);
bat.append(bint,1714636915:int);
bat.append(bint,nil:int);
rint:bat[:int] := pyapi.eval(nil:ptr,"return(arg1)",bint);
io.print(rint);

# long with null value
blng:= bat.new(:lng);
bat.append(blng,1804289383L);
bat.append(blng,846930886L);
bat.append(blng,1681692777L);
bat.append(blng,1714636915L);
bat.append(blng,nil:lng);
rlng:bat[:int] := pyapi.eval(nil:ptr,"return(arg1)",blng);
io.print(rlng);

# float with null value
bflt:= bat.new(:flt);
bat.append(bflt,18042.89383:flt);
bat.append(bflt,846.930886:flt);
bat.append(bflt,16.81692777:flt);
bat.append(bflt,1714636.915:flt);
bat.append(bflt,nil:flt);
rflt:bat[:flt] := pyapi.eval(nil:ptr,"return(arg1)",bflt);
io.print(rflt);

# double with null value
bdbl:= bat.new(:dbl);
bat.append(bdbl,180428.9383:dbl);
bat.append(bdbl,84.6930886:dbl);
bat.append(bdbl,168169.2777:dbl);
bat.append(bdbl,17146369.15:dbl);
bat.append(bdbl,nil:dbl);
rdbl:bat[:dbl] := pyapi.eval(nil:ptr,"return(arg1)",bdbl);
io.print(rdbl);

# convert int to string, then convert string to int
bint:= bat.new(:int);
bat.append(bint,1804289383:int);
bat.append(bint,846930886:int);
bat.append(bint,67:int);
bat.append(bint,124124124:int);
rstr:bat[:str] := pyapi.eval(nil:ptr,"return(arg1)",bint);
rint:bat[:int] := pyapi.eval(nil:ptr,"return([list(map(int, arg1))])",rstr);
io.print(rint);

