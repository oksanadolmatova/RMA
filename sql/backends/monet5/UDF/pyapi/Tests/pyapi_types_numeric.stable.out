stdout of test 'pyapi_types_numeric` in directory 'sql/backends/monet5/UDF/pyapi` itself:


# 19:41:49 >  
# 19:41:49 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=31934" "--set" "mapi_usock=/var/tmp/mtest-7261/.s.monetdb.31934" "--set" "monet_prompt=" "--forcemito" "--set" "mal_listing=2" "--dbpath=/usr/local/var/MonetDB/mTests_monetdb5_extras_pyapi" "--set" "mal_listing=2" "--set" "embedded_r=true" "--set" "embedded_py=true"
# 19:41:49 >  

# MonetDB 5 server v11.20.0
# This is an unreleased version
# Serving database 'mTests_monetdb5_extras_pyapi', using 8 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 64bit OIDs and 128bit integers dynamically linked
# Found 7.684 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2015 MonetDB B.V., all rights reserved
# Visit http://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://mytherin-N750JV:31934/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-7261/.s.monetdb.31934
# MonetDB/SQL module loaded
# MonetDB/Python module loaded
# MonetDB/R   module loaded

Ready.


#io.print(result);
#--------------------------#
# h	t  # name
# void	int  # type
#--------------------------#
[ 0@0,	20	]
[ 1@0,	40	]
[ 2@0,	60	]
[ 3@0,	80	]
#io.print(rbit);
#--------------------------#
# h	t  # name
# void	int  # type
#--------------------------#
[ 0@0,	1	]
[ 1@0,	0	]
[ 2@0,	1	]
[ 3@0,	0	]
[ 4@0,	nil	]
#io.print(rbte);
#--------------------------#
# h	t  # name
# void	int  # type
#--------------------------#
[ 0@0,	42	]
[ 1@0,	84	]
[ 2@0,	111	]
[ 3@0,	0	]
[ 4@0,	nil	]
#io.print(rsht);
#--------------------------#
# h	t  # name
# void	int  # type
#--------------------------#
[ 0@0,	42	]
[ 1@0,	82	]
[ 2@0,	0	]
[ 3@0,	240	]
[ 4@0,	nil	]
#io.print(rint);
#--------------------------#
# h	t  # name
# void	int  # type
#--------------------------#
[ 0@0,	1804289383	]
[ 1@0,	846930886	]
[ 2@0,	1681692777	]
[ 3@0,	1714636915	]
[ 4@0,	nil	]
#--------------------------#
# h	t  # name
# void	int  # type
#--------------------------#
[ 0@0,	1804289383	]
[ 1@0,	846930886	]
[ 2@0,	1681692777	]
[ 3@0,	1714636915	]
[ 4@0,	nil	]
#--------------------------#
# h	t  # name
# void	flt  # type
#--------------------------#
[ 0@0,	18042.895	]
[ 1@0,	846.9309	]
[ 2@0,	16.816927	]
[ 3@0,	1714636.9	]
[ 4@0,	nil	]
#io.print(rdbl);
#--------------------------#
# h	t  # name
# void	dbl  # type
#--------------------------#
[ 0@0,	180428.9383	]
[ 1@0,	84.6930886	]
[ 2@0,	168169.2777	]
[ 3@0,	17146369.15	]
[ 4@0,	nil	]
#io.print(rint);
#--------------------------#
# h	t  # name
# void	int  # type
#--------------------------#
[ 0@0,	1804289383	]
[ 1@0,	846930886	]
[ 2@0,	67	]
[ 3@0,	124124124	]

# 19:41:50 >  
# 19:41:50 >  "Done."
# 19:41:50 >  

